src/App.tsx

```diff
@@ -1,61 +1,131 @@
+import React, { useState } from 'react';
 import './App.scss';
 
 // import usersFromServer from './api/users';
 // import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number) {
+  const findUser = usersFromServer.find(user => user.id === userId);
+
+  return findUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [titleErrorMessage, setTitleErrorMessage] = useState('');
+  const [userErrorMessage, setUserErrorMessage] = useState('');
+  const [allTodos, setAllTodos] = useState(todos);
+
+  const handleTitleChange: React.ChangeEventHandler<HTMLInputElement>
+    = (event) => {
+      event.preventDefault();
+
+      setTitleErrorMessage('');
+
+      return setTitle(event.target.value);
+    };
+
+  const handleUserChange: React.ChangeEventHandler<HTMLSelectElement>
+    = (event) => {
+      event.preventDefault();
+
+      setUserErrorMessage('');
+
+      return setUserId(+event.target.value);
+    };
+
+  const handleSubmit: React.FormEventHandler = (event) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleErrorMessage('Please enter a title');
+    }
+
+    if (userId === 0) {
+      setUserErrorMessage('Please choose a user');
+    }
+
+    const findId = Math.max(...allTodos.map(todo => todo.id));
+
+    const todo: Todo = {
+      id: (findId + 1),
+      title,
+      userId: +userId,
+      completed: false,
+      user: getUser(+userId),
+    };
+
+    if (title && userId !== 0) {
+      setAllTodos([...todos, todo]);
```

Comment:
if you're setting new state value based on the previous one use callback  inside setState